File: leetcode/python/1150_check_if_number_is_majority.py
from bisect import bisect_left, bisect_right


class Solution:
    def isMajorityElement(self, nums: list[int], target: int) -> bool:
        l = bisect_left(nums, target)
        r = bisect_right(nums, target)
        return r - l > len(nums) // 2

File: leetcode/rust/src/1150_check_if_number_is_majority.rs
pub fn is_majority_element(nums: Vec<i32>, target: i32) -> bool {
    if nums[0] > target || nums[nums.len() - 1] < target {
        return false;
    }

    let r = nums.partition_point(|x| *x <= target);
    let l = nums.partition_point(|x| *x < target);
    r - l > nums.len() / 2
}